test/index.js

```diff
@@ -60,6 +60,44 @@ describe('serve(root)', function(){
       })
     })
 
+    describe('when maxage set', function(){
+      it('should add cache-control header', function(done){
+        var app = koa();
+
+        app.use(serve('test/fixtures', {
+          maxage: 3600 * 1000
+        }));
+
+        request(app.listen())
+            .get('/hello.txt')
+            .expect('Cache-Control', 'max-age=3600')
+            .expect(200, done);
+      })
+      it('can override cache-control header per path', function(done){
+        var app = koa();
+
+        app.use(serve('test/fixtures', {
+          updateOpts: function(path, opts) {
+            opts.maxage = path === '/hello.txt' ? 0 : 3600 * 1000;
```

Comment:
Maybe you can use [koajs/sendfile](https://github.com/koajs/sendfile) to send special file, or different `koa-static` instance.